pkg/controller/pcideviceclaim/pcideviceclaim_controller.go

```diff
@@ -133,20 +133,28 @@ func (h Handler) reconcilePCIDeviceClaims(hostname string) error {
 		// Check if PCI Device is already enabled for passthrough, but has no pre-existing PDC,
 		// if so, unbind the device (to force the user to make a proper PDC)
 		_, found := pdcNames[nodeAddr]
-		if !found && pd.Status.KernelDriverInUse == "vfio-pci" && hostname == pd.Status.NodeName {
+		if !found && pd.Status.KernelDriverInUse == "vfio-pci" && nodename == pd.Status.NodeName {
 			logrus.Infof("PCI Device %s is bound to vfio-pci but has no Claim, attempting to unbind", pd.Status.Address)
 			err = unbindPCIDeviceFromVfioPCIDriver(pd.Status.Address)
 			if err != nil {
 				return err
 			}
 		}
 		// After reboot, the PCIDeviceClaim will be there but the PCIDevice won't be bound to vfio-pci
-		if found && pd.Status.KernelDriverInUse != "vfio-pci" && hostname == pd.Status.NodeName {
+		if pd.Status.KernelDriverInUse != "vfio-pci" && nodename == pd.Status.NodeName {
 			// Set PassthroughEnabled to false
 			for _, pdc := range pdcs.Items {
 				if pdc.Spec.Address == pd.Status.Address {
 					logrus.Infof("Passthrough disabled for device %s", pd.Name)
 					pdc.Status.PassthroughEnabled = false
+					err = unbindPCIDeviceFromDriver(pd.Status.Address, pd.Status.KernelDriverInUse)
+					if err != nil {
+						logrus.Errorf("Error unbinding device after reboot: %s", err)
+					}
+					err = addNewIdToVfioPCIDriver(pd.Status.VendorId, pd.Status.DeviceId)
+					if err != nil {
+						logrus.Errorf("Error rebinding device after reboot: %s", err)
+					}
```

Comment:
should we set `PassthroughEnabled = true` here?

Reply:
Yes, good catch